dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/merge/dataelement/handler/TrackerDataElementMergeHandler.java

```diff
@@ -203,24 +203,25 @@ public void handleProgramRuleAction(List<DataElement> sources, DataElement targe
   }
 
   /**
-   * Method retrieving {@link Event}s by source {@link DataElement} references present in their
-   * {@link EventDataValue}s. All retrieved {@link Event}s will have their {@link DataElement} ref
-   * in {@link EventDataValue}s replaced with the target {@link DataElement}.
+   * Method retrieving {@link TrackerEvent}s by source {@link DataElement} references present in
+   * their {@link EventDataValue}s. All retrieved {@link TrackerEvent}s will have their {@link
+   * DataElement} ref in {@link EventDataValue}s replaced with the target {@link DataElement}.
    *
    * <p>A native query to retrieve events is required here as Hibernate does not support json
    * functions in the query. Because of this, all events are then updated at the end of this method,
    * which should prevent inconsistent state between Hibernate/JPA
    *
-   * @param sources source {@link DataElement}s used to retrieve {@link Event}s
-   * @param target {@link DataElement} which will be set as the {@link DataElement} in {@link Event}
-   *     eventDataValues
+   * @param sources source {@link DataElement}s used to retrieve {@link TrackerEvent}s
+   * @param target {@link DataElement} which will be set as the {@link DataElement} in {@link
+   *     TrackerEvent} eventDataValues
    * @param request merge request
    */
   public void handleEventDataValues(
       List<DataElement> sources, DataElement target, MergeRequest request) {
     Set<UID> sourceDeUids = UID.of(sources.toArray(new DataElement[0]));
     DataMergeStrategy mergeStrategy = request.getDataMergeStrategy();
 
+    // TODO(DHIS2-19702): Should we consider single events
```

Comment:
I think we should handle single events here as well. Since data elements could be either for Tracker or for Single Events.